libs/libc/pthread/pthread_barrierwait.c

```diff
@@ -90,14 +91,14 @@ int pthread_barrier_wait(FAR pthread_barrier_t *barrier)
 
   /* Disable pre-emption throughout the following */
 
-  sched_lock();
+  flags = enter_critical_section();
```

Comment:
libc can't call enter_critical_section directly since enter_critical_section belong to kernel space. we need either move the source code to sched/ or add mutex.